src/3rd_party_bundle_add.c

```diff
@@ -94,6 +96,136 @@ static bool parse_options(int argc, char **argv)
 	return true;
 }
 
+static bool is_binary(const char *filename)
+{
+	if (strncmp(filename, "/bin", strlen("/bin")) == 0 ||
+	    strncmp(filename, "/usr/bin", strlen("/usr/bin")) == 0 ||
+	    strncmp(filename, "/usr/local/bin", strlen("/usr/local/bin")) == 0) {
+		return true;
+	}
+
+	return false;
+}
+
+static enum swupd_code create_wrapper_script(const char *filename)
+{
+	enum swupd_code ret_code = 0;
+	FILE *fp = NULL;
+	char *bin_directory = NULL;
+	char *script = NULL;
+	char *binary = NULL;
+	char *env_path = NULL;
+	char *env_ld_path = NULL;
+	char *third_party_bin_path = NULL;
+	char *third_party_ld_path = NULL;
+	mode_t mode = 0755;
+
+	/* the path_prefix has the path to the 3rd-party repo, like:
+	 *  /<original_path_prefix>/opt/3rd-party/bundles/<repo>/
+	 *  we can use it to find the path to the bin directory:
+	 *  /<original_path_prefix>/opt/3rd-party/bin */
+	char *tmp1 = sys_dirname(globals.path_prefix);
+	char *tmp2 = sys_dirname(tmp1);
+	bin_directory = sys_path_join(tmp2, "/bin");
+	free_string(&tmp1);
+	free_string(&tmp2);
+
+	script = sys_path_join(bin_directory, sys_basename(filename));
+	binary = sys_path_join(globals.path_prefix, filename);
+
+	/* if a file already exists, report the problem to the user */
+	if (sys_file_exists(script)) {
+		warn("There is already a binary called %s in %s, it will be skipped\n", sys_basename(filename), bin_directory);
```

Comment:
Having a conflict notification at this point seems a little late. Couldn't we catch this earlier in the bundle-add instead and not actually add the bundle in that case?

Reply:
sure it can be done @bryteise . So the desired behavior would be that if there is a conflict inform the user and skip adding the bundle, correct?

Reply:
Yes, I'd expect we'd not add the bundle in that case since it would only be partially installed without the symlink.

Reply:
gotcha, I'll make the change, thanks